tests/grouping/test_pauli_group.py

```diff
@@ -0,0 +1,198 @@
+            PauliZ("b"),
+            PauliZ("a"),
+            PauliZ("a") @ PauliZ("b"),
+            PauliX("b"),
+            PauliY("b"),
+            PauliZ("a") @ PauliX("b"),
+            PauliZ("a") @ PauliY("b"),
+            PauliX("a"),
+            PauliX("a") @ PauliZ("b"),
+            PauliY("a"),
+            PauliY("a") @ PauliZ("b"),
+            PauliX("a") @ PauliX("b"),
+            PauliX("a") @ PauliY("b"),
+            PauliY("a") @ PauliX("b"),
+            PauliY("a") @ PauliY("b"),
+        ]
+
+        pg_2 = list(pauli_group(2, wire_map=wire_map))
+        assert all([expected.compare(obtained) for expected, obtained in zip(expected_pg_2, pg_2)])
+
+    @pytest.mark.parametrize(
+        "pauli_word_1,pauli_word_2,wire_map,expected_product",
+        [
+            (PauliX(0), Identity(0), {0: 0}, PauliX(0)),
+            (PauliZ(0), PauliY(0), {0: 0}, PauliX(0)),
+            (PauliZ(0), PauliZ(0), {0: 0}, Identity(0)),
+            (Identity("a"), Identity("b"), None, Identity("a")),
+            (PauliZ("b") @ PauliY("a"), PauliZ("b") @ PauliY("a"), None, Identity("b")),
+            (
+                PauliZ("b") @ PauliY("a"),
+                PauliZ("b") @ PauliY("a"),
+                {"b": 0, "a": 1},
+                Identity("b"),
+            ),
+            (
+                PauliZ(0) @ PauliY(1),
+                PauliX(0) @ PauliZ(1),
+                {0: 0, 1: 1},
+                PauliY(0) @ PauliX(1),
+            ),
+            (PauliZ(0) @ PauliY(1), PauliX(1) @ PauliY(0), {0: 0, 1: 1}, PauliX(0) @ PauliZ(1)),
+            (
+                PauliZ(0) @ PauliY(3) @ PauliZ(1),
+                PauliX(1) @ PauliX(2) @ PauliY(0),
+                None,
+                PauliX(0) @ PauliY(1) @ PauliX(2) @ PauliY(3),
+            ),
+            (
+                PauliX(0) @ PauliX(2),
+                PauliX(0) @ PauliZ(2),
+                None,
+                PauliY(2),
+            ),
+            (PauliZ("a"), PauliX("b"), {"a": 0, "b": 1}, PauliZ("a") @ PauliX("b")),
+            (
+                PauliZ("a"),
+                PauliX("e"),
+                {"a": 0, "b": 1, "c": 2, "d": 3, "e": 4},
+                PauliZ("a") @ PauliX("e"),
+            ),
+            (PauliZ("a"), PauliY("e"), None, PauliZ("a") @ PauliY("e")),
+            (
+                PauliZ(0) @ PauliZ(2) @ PauliZ(4),
+                PauliZ(0) @ PauliY(1) @ PauliX(3),
+                None,
+                PauliZ(2) @ PauliY(1) @ PauliZ(4) @ PauliX(3),
+            ),
+            (
+                PauliZ(0) @ PauliZ(4) @ PauliZ(2),
+                PauliZ(0) @ PauliY(1) @ PauliX(3),
+                {0: 0, 2: 1, 4: 2, 1: 3, 3: 4},
+                PauliZ(2) @ PauliY(1) @ PauliZ(4) @ PauliX(3),
+            ),
+            (
+                PauliZ(0) @ PauliY(3) @ PauliZ(1),
+                PauliX(1) @ PauliX(2) @ PauliY(0),
+                None,
+                PauliX(0) @ PauliY(3) @ PauliY(1) @ PauliX(2)
```

Comment:
New case that ensures that mixed up qubit orderings when no wire map is specified get constructed using combined set of wires in the order presented.